kbase-extension/static/kbase/js/util/jobLogViewer.js

```diff
@@ -341,12 +342,14 @@ define([
             this.lastJobState = {};
             this.listenersByType = {};
             this.lastMode = null;
-            this.looping = false; // if true, the log viewer is automatically requesting log updates
-            this.stopped = false; // if true, we have stopped automatically fetching logs
-            this.listeningForJob = false; // if true, this means we're listening for job updates
-            this.awaitingLog = false; // if true, there's a log request fired that we're awaiting
             this.requestLoop = null; // the timeout object
-            this.scrollToEndOnNext = false;
+            this.state = {
```

Comment:
put all this stuff to do with maintaining state in an appropriately-named attribute.